shm/game.cpp

```diff
@@ -0,0 +1,380 @@
+    while (objectToSell->getAmount() < amountOfCargoToSell)
+    {
+        std::cout << "Captain, we don't have so many of " << objectToSell->getName() << " What we do now?\n1. Change amount\n2. Exit" << '\n';
+        std::cin >> sellOption;
+        if (sellOption == 1)
+        {
+            std::cout << "Amount of " << objectToSell->getName() << " to sell:\n";
+            std::cin >> amountOfCargoToSell;
+        }
+        else
+        {
+            return;
+        }
+    }
+    atStore = std::find_if(actualStoreVector.begin(), actualStoreVector.end(), [&objectToSell](const auto &atStore)
+                           { return objectToSell->getName() == atStore->getName(); });
+    ship_->unload(objectToSell, amountOfCargoToSell);
+    player_->setMoney(player_->getMoney() + (*atStore)->getPrice() * amountOfCargoToSell);
+    map_.getIsland(player_->getPlayerPosition())->getStore()->addCargo(objectToSell, amountOfCargoToSell);
+    time_->onTimeChange();
+}
+
+void Game::printMap(const std::shared_ptr<Player> player)
+{
+    int currentIsland = 0;
+    for (auto island : map_.getEveryIsland())
+    {
+        currentIsland++;
+        if (island.getPosition() == player_->getPlayerPosition())
+        {
+            std::cout << "We are on Island " << currentIsland << '\n';
+            return;
+        }
+    }
+    std::cout << "We are on the sea!" << '\n';
+}
+
+void Game::travel()
+{
+    std::cout << "Where you want to travel?\n";
+    int islandCounter = 0;
+    std::vector<int> daysToGo;
+    daysToGo.reserve(map_.getEveryIsland().size());
+    for (auto island : map_.getEveryIsland())
+    {
+        int amountOfDays = island.getPosition().distance(player_->getPlayerPosition()) / ship_->getSpeed();
+        if (amountOfDays == 0 && !(island.getPosition() == player_->getPlayerPosition()))
+        {
+            daysToGo.push_back(1);
+        }
+        else
+        {
+            daysToGo.push_back(amountOfDays);
+        }
+        islandCounter++;
+        std::cout << "Island " << islandCounter << " Travel time: " << daysToGo[islandCounter - 1] << " days\n";
+    }
+    std::cout << ++islandCounter << ". Exit\n";
+    std::cout << "Where are we going captain?: ";
+    islandCounter = 0;
+    std::cin.clear();
+    std::cin >> islandCounter;
+    while (islandCounter > map_.getEveryIsland().size() + 1)
+    {
+        std::cout << "Captain you have old map! Here is new one, so where we are going?: \n";
+        std::cin.clear();
+        std::cin >> islandCounter;
+    }
+    if (islandCounter == map_.getEveryIsland().size() + 1)
+    {
+        std::cout << "I changed my mind, we stay here!\n";
+        return;
+    }
+    player_->setPlayerPosition(map_.getEveryIsland()[islandCounter - 1].getPosition().getPositionX(), map_.getEveryIsland()[islandCounter - 1].getPosition().getPositionY());
+    for (int i = 0; i < daysToGo[islandCounter - 1]; i++)
+    {
+        time_->onTimeChange();
+    }
```

Comment:
You can also create method `travell(int days)` and it will be more precise waht happened here :)